design/ironic-standalone-networking-service.md

```diff
@@ -0,0 +1,734 @@
+
+```go
+// SwitchCredentialType defines the type of credential used for switch authentication.
+type SwitchCredentialType string
+
+const (
+    // SwitchCredentialTypePassword indicates password-based authentication
+    SwitchCredentialTypePassword SwitchCredentialType = "password"
+    // SwitchCredentialTypePublicKey indicates SSH public key-based authentication
+    SwitchCredentialTypePublicKey SwitchCredentialType = "publicKey"
+)
+
+// SwitchCredentials defines the credentials used to access the switch.
+type SwitchCredentials struct {
+    // Type specifies the type of credential used for authentication.
+    // +kubebuilder:validation:Enum=password;publicKey
+    Type SwitchCredentialType `json:"type"`
+
+    // The name of the secret containing the switch credentials.
+    // For password authentication, requires key "password".
+    // For SSH key authentication, requires keys "username", "privateKey", and optionally "publicKey".
+    SecretName string `json:"secretName"`
+}
+
+// BareMetalSwitchSpec defines the desired state of BareMetalSwitch.
+type BareMetalSwitchSpec struct {
+    // Important: Run "make generate manifests" to regenerate code
+    // after modifying this file
+
+    // Address is the network address of the TOR switch (IP address or hostname).
+    // +kubebuilder:validation:Required
+    Address string `json:"address"`
+
+    // MACAddress is the MAC address of the switch management interface.
+    // +kubebuilder:validation:Pattern=`[0-9a-fA-F]{2}(:[0-9a-fA-F]{2}){5}`
+    MACAddress string `json:"macAddress,omitempty"`
+
+    // DriverType specifies the type of driver to use for switch management.
+    // Examples: "cisco", "arista", "juniper", "dell", "generic"
+    // +kubebuilder:validation:Required
+    DriverType string `json:"driverType"`
+
+    // DeviceType specifies the specific device type or model of the switch.
+    // This helps with driver-specific functionality.
+    DeviceType string `json:"deviceType,omitempty"`
+
+    // Credentials defines how to authenticate with the switch.
+    // +kubebuilder:validation:Required
+    Credentials SwitchCredentials `json:"credentials"`
+
+    // Port specifies the management port to connect to (e.g., SSH port 22, HTTPS port 443).
+    // If not specified, the driver will use its default port.
+    // +kubebuilder:validation:Minimum=1
+    // +kubebuilder:validation:Maximum=65535
+    // +optional
+    Port *int32 `json:"port,omitempty"`
+
+    // SkipTLSVerify disables verification of server certificates when using
+    // HTTPS to connect to the switch. This is required when the server
+    // certificate is self-signed, but is insecure because it allows a
+    // man-in-the-middle to intercept the connection.
+    // +optional
+    SkipTLSVerify *bool `json:"skipTLSVerify,omitempty"`
+}
+
+type BareMetalSwitchStatus struct {
+    // Conditions track the state of the switch configuration
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
```

Comment:
Which conditions are we going to expose?

Reply:
I won't lie... I think I put this in out of habit and don't really have a clear idea here.   Was probably thinking we could possibly try accessing the switch to test the credentials, but might be overkill in a first implementation.